packages/astro/src/core/config/timer.ts

```diff
@@ -0,0 +1,65 @@
+import fs from 'fs';
+
+// Type used by `bench-memory.js`
+export interface Stat {
+	elapsedTime: number;
+	heapUsedChange: number;
+	heapUsedTotal: number;
+}
+
+interface OngoingStat {
+	startTime: number;
+	startHeap: number;
+}
+
+/**
+ * Timer to track certain operations' performance. Used by Astro's scripts only.
+ * Set `process.env.ASTRO_TIMER_PATH` truthy to enable.
+ */
+export class AstroTimer {
+	private enabled: boolean;
+	private ongoingTimers: Map<string, OngoingStat> = new Map();
```

Comment:
Maybe we could use `WeakMap`? It doesn't seem we iterate them

Reply:
Weak maps only allows objects as keys, we also need to refetch the `OngoingStat` when calling the `end()` method, so we need it to persists indefinitely as there isn't a reference to pass around.